Update Finding and Choosing Packages section (#2104)
NuGet/docs.microsoft.com-nuget: docs/consume-packages/Finding-and-Choosing-Packages.md

## Finding packages

> When you visit nuget.org or open the Package Manager UI in Visual Studio, you see a list of packages sorted by relevancy. This immediately shows you the most widely-used packages across the millions of .NET projects. There's a good chance, then, that at least some of the packages listed on the first few pages will be useful in your projects.

Reviewer: What do you think of improving this section (sorry, I know this is a little out-of-scope for your additions! 😅)

```suggestion
When you visit nuget.org or open the Package Manager UI in Visual Studio, you see a list of packages sorted by relevancy. This shows you the most widely used packages across all .NET projects. There's a good chance that some of these packages may be useful in your projects.
```

![Default view of nuget.org/packages showing the most popular packages](media/Finding-01-Popularity.png)

> Notice the **Filter** button on the upper right of the page. When clicked, you will expand the Advanced Search panel which will present you with more precise soriting and filtering options.

Reviewer: ```suggestion
Notice the **Filter** button on the upper right of the page. When clicked, the Advanced Search panel expands to present more precise sorting and filtering options.
```

![Search results for 'json' on nuget.org](media/Finding-02-SearchResults.png)

You can use the **Package type** filter to only show packages of a specific type.
To sort the search results, you can use the **Sort by** section of the panel to select your preferred method of sorting:
- **`Relevance`**: This is the default behavior. It sorts results according to an internal scoring algorithm.

> - **`Downloads`**: Sorts the search results by the total number of downloads, starting with the higher download count.

Reviewer: What do you think of this?

```suggestion
- **`Downloads`**: Sorts the search results by the total number of downloads in descending order.
```

- **`Recently updated`**: Sorts the search results by their latest version's creation date, in descending chronological order.

In the **Options** section, we can find the **`Include prerelease`** checkbox.

> When checked, nuget.org shows all versions of packages including beta and other early releases. To show only stable released, clear the option.

Reviewer: ```suggestion
When checked, nuget.org shows all versions of packages including pre-releases. To show only stable versions, clear the option.
```

Reviewer: One may use pre-releases for things other than early releases, hence why I'd lean towards more general phrasing

To apply the search filters, click on the **`Apply`** button. You can always get back to the default behavior by clicking on the **`Reset`** button.

> For specific needs, searching by tags (within the Visual Studio Package Manager or on a portal like nuget.org) is the most common means of discovering a suitable package. For example, searching on "json" lists all NuGet packages that are tagged with that keyword and thus have some relationship to the JSON data format.
> 
> You can also search using the package ID, if you know it. See [Search Syntax](#search-syntax) below.

Reviewer: I know you didn't modify this, but what do you think of rewriting it a little bit?

```suggestion
You can also use the [search syntax](#search-syntax) to filter on tags, owners, and package ids.
```

## Finding packages

When you visit nuget.org or open the Package Manager UI in Visual Studio, you see a list of packages sorted by relevancy. This shows you the most widely used packages across all .NET projects. There's a good chance that some of these packages may be useful for your own projects!

![Default view of nuget.org/packages showing the most popular packages](media/Finding-01-Popularity.png)

> Notice the **Filter** button on the upper right of the page. When clicked, the Advanced Search panel expands to present sorting and filtering options.

Reviewer 2: Previous paragraph mentions both PM UI and nuget.org but Filter is on nuget.org. To be clear, maybe say "On nuget.org, notice the..."

![Search results for 'json' on nuget.org](media/Finding-02-SearchResults.png)

> You can use the **Package type** filter to only show packages of a specific type:

Reviewer: "Only" feels a little redundant to me here. What do you think of this?

```suggestion
You can use the **Package type** filter to show packages of a specific type:
```

> - **`All types`**: This is the default behavior. It shows all packages regarding of their type.

Reviewer: ```suggestion
- **`All types`**: This is the default behavior. It shows all packages regardless of their type.
```

> - **`Dependency`**: This only shows NuGet dependency packages which represent libraries.

Reviewer: I'm not sure the libraries distinction is all that clear to our customers. What do you think of something like this?

```suggestion
- **`Dependency`**: Regular NuGet packages that can be installed into your project.
```

> - **`.NET tool`**: This filters to .NET tools. For more info: [How to manage .NET Core tools](https://docs.microsoft.com/en-us/dotnet/core/tools/global-tools)

Reviewer: Let's give a quick summary of what this package type does. What do you think of something like this?

```suggestion
- **`.NET tool`**: This filters to [.NET tools](https://docs.microsoft.com/en-us/dotnet/core/tools/global-tools), a NuGet package that contains a console application.
```

> - **`Template`**: This filters to packages that are project and item templates. For more info: [manage .NET project and item templates](https://docs.microsoft.com/en-us/dotnet/core/install/templates?pivots=os-windows)

Reviewer: I believe we have magical link URL syntax to other Microsoft docs websites:

```suggestion
- **`Template`**: This filters to packages that are project and item templates. For more info: [manage .NET project and item templates](/dotnet/core/install/templates)
```

Reviewer: Same comment for line 27

> - **`Template`**: This filters to packages that are project and item templates. For more info: [manage .NET project and item templates](https://docs.microsoft.com/en-us/dotnet/core/install/templates?pivots=os-windows)

Reviewer: What do you think of something like this?

```suggestion
- **`Template`**: This filters to [.NET templates](/dotnet/core/install/templates), which can be used to create new projects using the [`dotnet new`](/dotnet/core/tools/dotnet-new) command. 
```

Author: I don't know if we should give the command here. Also the wording "packages that are project and item templates" was directly copied from the .NET Template doc and I think it's more clear

Reviewer: Wording like `.. to create new projects and items...` seems reasonable to me too 👍

What are reasons against providing the `dotnet new` command? Templates entirely revolve the `dotnet new` command so I think this helps provide context to customers

> To sort the search results, you can use the **Sort by** section of the panel to select your preferred method of sorting:

Reviewer: This is a bit wordy. What do you think of this?

```suggestion
You can use the **Sort by** option to sort the search results:
```

- **`Relevance`**: This is the default behavior. It sorts results according to an internal scoring algorithm.

> - **`Downloads`**: Sorts the search results by the total number of downloads in descending order.
> - **`Recently updated`**: Sorts the search results by their latest version's creation date, in descending chronological order.

Reviewer: We should make the commas consistent for the last two bullet points. Either:

```suggestion
- **`Downloads`**: Sorts the search results by the total number of downloads, in descending order.
- **`Recently updated`**: Sorts the search results by their latest version's creation date, in descending chronological order.
```

Or:

```suggestion
- **`Downloads`**: Sorts the search results by the total number of downloads in descending order.
- **`Recently updated`**: Sorts the search results by their latest version's creation date in descending chronological order.
```
